backend/CircleForms/ModelLayer/GamemodeService.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Threading.Tasks;
+using CircleForms.Database.Models.Posts.Enums;
+using CircleForms.Database.Services.Abstract;
+using CircleForms.ExternalAPI.OsuApi;
+using MongoDB.Bson;
+
+namespace CircleForms.ModelLayer;
+
+public class GamemodeService : IGamemodeService
+{
+    private readonly IUserRepository _usersService;
+    private readonly IOsuApiProvider _osuApiProvider;
+
+    private const string _statisticsKey = "Statistics";
+
+    public GamemodeService(IUserRepository usersService, IOsuApiProvider osuApiProvider)
+    {
+        _usersService = usersService;
+        _osuApiProvider = osuApiProvider;
+    }
+
+    public async Task<Result<BsonDocument>> GetStatistics(string userId, Gamemode mode)
+    {
+        var user = await _usersService.Get(userId);
+        if (user is null)
+        {
+            return Result<BsonDocument>.NotFound(userId);
+        }
+
+        var modeName = Enum.GetName(mode);
+
+        if (!user.Osu.Contains(_statisticsKey))
+        {
+            user.Osu.Add(new BsonElement(_statisticsKey, new BsonDocument()));
```

Comment:
`_logger.Critical` or `Error` if no statistics present, it should never happen

Reply:
It will always happen when we have a new user. By default `Statistics` doesn't exist

Reply:
oh yeah kinda true